invenio_rdm_records/utils.py

```diff
@@ -8,10 +8,33 @@
 
 """Utils for Invenio RDM Records."""
 
-from flask import current_app
+from flask import current_app, request
+from copy import deepcopy
 
 
 def is_doi_locally_managed(doi_value):
     """Determine if a DOI value is locally managed."""
     return any(doi_value.startswith(prefix) for prefix in
                current_app.config['RDM_RECORDS_LOCAL_DOI_PREFIXES'])
+
+
+class DynamicAggregation(dict):
+    """Aggregation that is fully resolved in the request context."""
+
+    def __init__(self, base_agg, function_agg):
```

Comment:
You can match the [`dict(mapping, **kwargs)`](https://docs.python.org/3/library/stdtypes.html#dict) constructor, and basically do something like:

```suggestion
    def __init__(self, agg, func=None, **kwargs):
```